python_modules/dagster/dagster/_core/definitions/load_assets_from_modules.py

```diff
@@ -23,14 +36,22 @@
 from dagster._core.errors import DagsterInvalidDefinitionError
 
 
-def find_objects_in_module_of_types(module: ModuleType, types) -> Iterator:
-    """Yields objects of the given type(s)."""
+def find_objects_in_module_of_types(
+    module: ModuleType,
+    types: Union[Type, Tuple[Type, ...]],
+    mode: Literal["instance", "subclass"] = "instance",
+) -> Iterator:
+    """Yields instances or subclasses of the given type(s)."""
     for attr in dir(module):
         value = getattr(module, attr)
-        if isinstance(value, types):
-            yield value
-        elif isinstance(value, list) and all(isinstance(el, types) for el in value):
-            yield from value
+        if mode == "instance":
+            if isinstance(value, types):
+                yield value
+            elif isinstance(value, list) and all(isinstance(el, types) for el in value):
+                yield from value
+        elif mode == "subclass" and isinstance(value, type):
+            if issubclass(value, types):
+                yield value
```

Comment:
ew this `mode` thing is fishy. Just having two separate methods for the two use cases, is better IMO